README.md

```diff
@@ -1,21 +1,319 @@
 
-## Understand this workspace
 
-Run `nx graph` to see a diagram of the dependencies of the projects.
+<details id="context">
+  <summary><b>Context</b></summary>
+
+1. The Hasura Console can be loaded in six different "types", and two different "modes" (12 cases to manage). This is messy represented by the `window.__env` object the Hasura finds when loads.
+2. When the Console is launched by the CLI server, we must care about [the old CLI server and the new one](https://github.com/hasura/graphql-engine-mono/pull/7256#discussion_r1071239279) (the cases to manage grows to 18).
+3. There are additional things to consider that are retrieved dynamically
+   1. Lux entitlements
+   2. Pricing tiers
+   3. EE Trial license
+   4. ... and that's just the beginning...
+
+*(here is a [good summary by @beaussan](https://hasurahq.slack.com/archives/C0423EHN686/p1676297462152259?thread_ts=1676291643.033329&cid=C0423EHN686) about the current situation).*
+
+All of the abovementioned points concur in showing some features or not, in trying to upsell Hasura to the customers, etc.
+</details>
+
+<details id="the-problem">
+  <summary><b>The problem</b></summary>
+
+The mess with `window.__env` and the above cases is already high, introducing a lot of bugs and PRs in the Console.
+
+The engineers working on the Console not only need to understand when a feature is enabled or not but also to understand why in order to offer the best possible UX (in upselling terms too) to the customers.
+</details>
+
+
+<details id="how-the-problem-is-tackled-right-now">
+  <summary><b>How the problem is tackled right now</b></summary>
+
+1. Through [the `proConsole` module](https://github.com/hasura/graphql-engine-mono/blob/main/frontend/libs/console/legacy-ce/src/lib/utils/proConsole.ts) where we have been hidden checking the console type/mode in the last six months.
+2. Through using the above utilities combined with custom checks, [here is an example](https://github.com/hasura/graphql-engine-mono/blob/734ac1a250ca8efed568ac15cd44fe384a30cf03/frontend/libs/console/legacy-ee/src/lib/components/Main/Main.js).
+</details>
+
+
+<details id="whats-the-platform-team-goal">
+  <summary><b>What's the Platform team goal?</b></summary>
+
+Essentially, to ease the developer life dealing with the above mess. This is possible through
+
+1. Centralizing manage all the data/info/properties/vars that impact what the Console shows
+2. Allow this centralized management to scale for future needs
+3. Hide all the ~mess~ **implementation details** of dealing with the `window.__env`, pricing plans, entitlements, etc.
+
+Steps and scope of this POC
+</details>
+
+
+
+## What to expect from this POC
+
+This POC **does not include**:
+- battle-tested code
+- Storybook utilities
+- definitive libraries and file system
+- some thorough real use cases the Console must deal with
+
+This POC **includes**:
+- a high-level proposal of the TS APIs needed to hide/show a feature
```

Comment:
**suggestion:** Maybe you should not use the *high-level* word as it suggests the API should be refined.